string type
string condition
---
ObjectPosition position